From 6fe7234b993647ed84ff74df4ca20f54f761bac1 Mon Sep 17 00:00:00 2001
From: Christian Stewart <christian@paral.in>
Date: Thu, 29 Oct 2020 02:13:56 -0700
Subject: [PATCH] fix(builder): wait briefly after build finishes for image tag

Sometimes Docker will reply that the build completed before finishing the
tagging.

Signed-off-by: Christian Stewart <christian@paral.in>
---
 builder/builder.go | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/builder/builder.go b/builder/builder.go
index 000786a..8c44979 100644
--- a/builder/builder.go
+++ b/builder/builder.go
@@ -97,7 +97,13 @@ func (b *Builder) build(buildPath string) error {
 		return <-res
 	}
 
-	return b.dockerBuild(dockerClient, buildPath, b.config.ImageName())
+	if err := b.dockerBuild(dockerClient, buildPath, b.config.ImageName()); err != nil {
+		return err
+	}
+
+	// race: briefly for image tag to complete
+	<-time.After(time.Millisecond * 200)
+	return nil
 }
 
 // build builds the dockerfile in a directory.
-- 
2.29.0

